qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -676,12 +676,14 @@ def _dimension_limits(
     :param limits: An optional Sequence of length 2 containing limits for plotting dimensions. If
                     not provided the full extent of each dimension is used.
     """
-    dim_limits = [(dim.getMinimum(), dim.getMaximum()) for dim in [workspace.getDimension(i) for i in range(workspace.getNumDims())]]
+    dims = [workspace.getDimension(i) for i in range(workspace.getNumDims())]
+    dim_limits = [(dim.getMinimum(), dim.getMaximum()) for dim in dims]
+    dim_width = [dim.getBinWidth() for dim in dims]
```

Comment:
I suppose this comprehension could probably be done in the main loop - i.e. loop over `idim in rnage(workspace.getNumDims())` -  given we are looping over the dimensions anyway, but as there's only usually 2-4 dimensions this is pretty negligible!

Reply:
Hmm, why do you think the test should fail? the test doesn't cover the current new addition as `bin_params` is not None and the latest addition handles the case when `bin_params` is None

Ok, I will refactor the code, thanks for spotting the duplication

Reply:
Thanks, I see why the test didn't fail, but I think that indicates we need another test for when `bin_params` is `None`